storage/elasticsearch/client.go

```diff
@@ -83,34 +83,43 @@ type ElasticSearchClientInterface interface {
 	Delete(obj string, id string) (elastigo.BaseResponse, error)
 	BulkDelete(obj string, id string)
 	Search(obj string, query string, index string) (elastigo.SearchResult, error)
-	Start(name string, mappings []map[string][]byte, limits ElasticLimits)
+	Start()
 	GetIndexAlias() string
 	GetIndexAllAlias() string
 }
 
+// Mappings describes the mappings of the clinet connection
+type Mappings []map[string][]byte
+
 // ElasticIndex describes an ElasticSearch index and its current status
 type ElasticIndex struct {
 	sync.Mutex
 	entriesCounter int
-	mappings       []map[string][]byte
-	name           string
 	path           string
 	timeCreated    time.Time
-	limits         ElasticLimits
+}
+
+// ConnConfig describes the ElasticSearch configuration section
+type ConnConfig struct {
+	ElasticHost  string
+	MaxConns     int
+	RetrySeconds int
+	BulkMaxDocs  int
+	BulkMaxDelay int
 }
 
 // ElasticSearchClient describes a ElasticSearch client connection
 type ElasticSearchClient struct {
-	connection   *elastigo.Conn
-	indexer      *elastigo.BulkIndexer
-	started      atomic.Value
-	quit         chan bool
-	wg           sync.WaitGroup
-	index        *ElasticIndex
-	maxConns     int
-	retrySeconds int
-	bulkMaxDocs  int
-	bulkMaxDelay int
+	connection *elastigo.Conn
+	indexer    *elastigo.BulkIndexer
+	started    atomic.Value
+	quit       chan bool
+	wg         sync.WaitGroup
+	name       string
+	mappings   Mappings
+	idxCfg     IndexConfig
```

Comment:
Please do not use short names, change idx to index, it is confusing. so for this variable use indexConfig

Reply:
And variables are typically shorter than structs, as a variable lives in a block or struct (constrained scope) while a struct lives within the module (much larger scope) ... thus in idiomatic golang (and other languages) variables tend to be shorter

Reply:
ok will change idxCfg --> indexCfg